### ISSUE TITLE
Runtime Error when computing  logdet

### ISSUE THREAD

**Issue by sndnyang** [tsb0601/EMP-SSL#1]
Hi
It's an amazing work and I really appreciate your paper and open source code.

However, I tried to train the model on my server and encountered an issue.

```
File "/home//research/empssl/main.py", line 162, in main
    loss_TCR = cal_TCR(z_proj, criterion, num_patches)
  File "/home//research/empssl/main.py", line 100, in cal_TCR
    loss += criterion(z_list[i])
  File "/home//software/anaconda3/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home//research/empssl/loss.py", line 90, in forward
    return - self.compute_discrimn_loss(X.T)
  File "/home//research/empssl/loss.py", line 86, in compute_discrimn_loss
    logdet = torch.logdet(I + scalar * W.matmul(W.T))
RuntimeError
......final some rows
extern "C"
__launch_bounds__(512, 4)
__global__ void reduction_prod_kernel(ReduceJitOp r){
  r.run();
}
nvrtc: error: invalid value for --gpu-architecture (-arch)
```

Do you have any suggestions for it? Maybe I will try it on Google Colab.

**Comment by sndnyang** [comment]
It works on Google Colab......
I need to figure out how to run on my server.

**Comment by Ferry7z** [comment]
Hey,have you finished the problem?I met the same.

**Comment by sndnyang** [comment]
> Hey,have you finished the problem?I met the same.

I think it's solved by latest PyTorch (or 2.0 )

**Comment by Ferry7z** [comment]
> > Hey,have you finished the problem?I met the same.
> 
> I think it's solved by latest PyTorch (or 2.0 )

I have used the latest version of all these pocket it need,and it didn't work.But still thank you.
